datashape/typesets.py

```diff
@@ -115,3 +120,70 @@ def __getitem__(self, key):
 real = TypeSet(*integral | floating, name='real')
 numeric = TypeSet(*integral | floating | complexes, name='numeric')
 scalar = TypeSet(*boolean | numeric, name='signed')
+
+
+supertype_map = {
+    int8: signed,
+    int16: signed,
+    int32: signed,
+    int64: signed,
+    uint8: unsigned,
+    uint16: unsigned,
+    uint32: unsigned,
+    uint64: unsigned,
+    float16: floating,
+    float32: floating,
+    float64: floating,
+    complex64: complexes,
+    complex128: complexes,
+    bool_: boolean
+}
+
+
+def supertype(measure):
+    """Get the super type of a concrete numeric type
+
+    Examples
+    --------
+    >>> supertype(int8)
+    {signed}
+
+    >>> supertype(float32)
+    {floating}
+
+    >>> supertype(complex128)
+    {complexes}
+
+    >>> supertype(bool_)
+    {boolean}
+    """
+    measure = getattr(measure, 'ty', measure)
```

Comment:
This seems easier to understand.  Actually, I'm not even sure why `ty` is the field name for this.  What does that stand for?
```
if isinstance(measure, Option):
    measure = measure.ty
```